driver-core/src/main/java/com/datastax/driver/core/RequestHandler.java

```diff
@@ -234,6 +241,23 @@ private void setFinalResult(
     }
   }
 
+  private void logServerWarnings(List<String> warnings) {
+    // log the warnings if they have NOT been disabled (double negative)
+    if (!Boolean.getBoolean(RequestHandler.DISABLE_QUERY_WARNING_LOGS)) {
+      // comma separated list of warnings
+      StringBuilder warningString = new StringBuilder();
+      for (String warning : warnings) {
+        warningString.append(warning).append(", ");
+      }
+      // truncate the statement query to the MAX_QUERY_LOG_LENGTH, if necessary
+      final String queryString = statement.toString();
+      logger.warn(
```

Comment:
I wonder if we shouldn't print one message per warning; this way logs can be more easily grep'ed or post-processed.